keras_cv/models/backbones/densenet/densenet_backbone_test.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+import tensorflow as tf
+from absl.testing import parameterized
+from tensorflow import keras
+
+from keras_cv.models.backbones.densenet.densenet_backbone import (
+    DenseNet121Backbone,
+)
+from keras_cv.models.backbones.densenet.densenet_backbone import (
+    DenseNetBackbone,
+)
+from keras_cv.utils.train import get_feature_extractor
+
+
+class DenseNetBackboneTest(tf.test.TestCase, parameterized.TestCase):
+    def setUp(self):
+        self.input_batch = tf.ones(shape=(2, 224, 224, 3))
+
+    def test_valid_call(self):
+        model = DenseNetBackbone(
+            blocks=[6, 12, 24, 16],
+            include_rescaling=False,
+        )
+        model(self.input_batch)
+
+    def test_valid_call_applications_model(self):
+        model = DenseNet121Backbone()
+        model(self.input_batch)
+
+    def test_valid_call_with_rescaling(self):
+        model = DenseNetBackbone(
+            blocks=[6, 12, 24, 16],
+            include_rescaling=True,
+        )
+        model(self.input_batch)
+
+    @parameterized.named_parameters(
+        ("tf_format", "tf", "model"),
+        ("keras_format", "keras_v3", "model.keras"),
+    )
+    def test_saved_model(self, save_format, filename):
+        model = DenseNetBackbone(
+            blocks=[6, 12, 24, 16],
+            include_rescaling=False,
+        )
+        model_output = model(self.input_batch)
+        save_path = os.path.join(self.get_temp_dir(), filename)
+        model.save(save_path, save_format=save_format)
+        restored_model = keras.models.load_model(save_path)
+
+        # Check we got the real object back.
+        self.assertIsInstance(restored_model, DenseNetBackbone)
+
+        # Check that output matches.
+        restored_output = restored_model(self.input_batch)
+        self.assertAllClose(model_output, restored_output)
+
+    @parameterized.named_parameters(
+        ("tf_format", "tf", "model"),
+        ("keras_format", "keras_v3", "model.keras"),
+    )
```

Comment:
Mark these tests "large" as in https://github.com/keras-team/keras-cv/pull/1804

Reply:
Done

Reply:
> Also don't forget to include #1806 style refactoring

Yep, made a note to self [here](https://github.com/keras-team/keras-cv/pull/1759#issuecomment-1563150799)